### PR TITLE
chore: add .editorconfig for formatting

### PR DESCRIPTION
**Pull request by n-r-b**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# http://editorconfig.org
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
```
Comment (geek-soft) [inline .editorconfig:6]: What do you think about using 2 spaces everywhere? I think it makes code more concise.

Comment (n-r-b) [reply .editorconfig:6]: i thought this was something that was agreed on previously, so left it in there as a default. But more than happy to use 2 spaces everywhere

Comment (n-r-b) [reply .editorconfig]: updated .editorconfig and applied formatting

Comment (geek-soft) [reply .editorconfig]: We agreed it need but I thought we could change it. Not forcing but I prefer 2 spaces generally.

**Pull request merged**